File: backend/src/main/java/tn/tawiniya/tounisiya/entity/TechnicienProfile.java
package tn.tawiniya.tounisiya.entity;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDate;
import java.time.LocalDateTime;
/**
 * Fiche personnelle complète d'un technicien, remplie par lui-même depuis son espace.
 * Un profil par utilisateur (relation 1-1).
 */
@Entity
@Table(name = "technicien_profiles")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class TechnicienProfile {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @OneToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "user_id", nullable = false, unique = true)
    private User user;
    // --- Identité ---
    private String nom;
    private String prenom;
    private LocalDate dateNaissance;
    private String cin;
    private String nomParent;
    private String adresse;
    // --- Contacts ---
    private String gsm;
    private String gsmParent;
    private String gsmBinome;
    private String email;
    private String facebook;
    private String tiktok;
    private String instagram;
    // --- Formation ---
    private String diplome;
    private String specialite;
    private String niveauScolaire;
    private boolean permisConduite;
    private LocalDate datePermis;
    // --- Contrat & administratif ---
    private String typeContrat;
    private String numeroCnss;
    private String numeroD17;
    private String numeroBanquePoste;
    // --- Santé ---
    @Enumerated(EnumType.STRING)
    private GroupeSanguin groupeSanguin;
    private Double poidsKg;
    private Double hauteurCm;
    private String pointureChaussure;
    private String tailleVetements;
    @Column(columnDefinition = "TEXT")
    private String maladiesChroniques;
    @Column(columnDefinition = "TEXT")
    private String allergies;
    @Column(columnDefinition = "TEXT")
    private String operations;
    private boolean tatouage;
    // --- Emploi ---
    private LocalDate dateEmbauche;
    private String experience;
    @Column(columnDefinition = "TEXT")
    private String societesEtPeriodes;
    private Double salaireDepart;
    private Integer nombreJoursConge;
    private String gsmSociete;
    // --- Documents & médias (chemins publics) ---
    private String photoProfilPath;
    private String photoCouverturePath;
    private String copieCinPath;
    private String copieExtraitNaissancePath;
    private String copieDiplomePath;
    private String copiePermisPath;
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;
    @PrePersist
    protected void onCreate() {
        LocalDateTime now = LocalDateTime.now();
        this.createdAt = now;
        this.updatedAt = now;
    }
    @PreUpdate
    protected void onUpdate() {
        this.updatedAt = LocalDateTime.now();
    }
}

File: frontend/src/types/technicienProfile.ts
export type GroupeSanguin =
  | 'A_POSITIF' | 'A_NEGATIF'
  | 'B_POSITIF' | 'B_NEGATIF'
  | 'AB_POSITIF' | 'AB_NEGATIF'
  | 'O_POSITIF' | 'O_NEGATIF';
export const GROUPE_SANGUIN_LABELS: Record<GroupeSanguin, string> = {
  A_POSITIF: 'A+',
  A_NEGATIF: 'A-',
  B_POSITIF: 'B+',
  B_NEGATIF: 'B-',
  AB_POSITIF: 'AB+',
  AB_NEGATIF: 'AB-',
  O_POSITIF: 'O+',
  O_NEGATIF: 'O-',
};
export interface TechnicienProfile {
  id: number;
  userId: number;
  userEmail: string;
  nom: string | null;
  prenom: string | null;
  dateNaissance: string | null;
  cin: string | null;
  nomParent: string | null;
  adresse: string | null;
  gsm: string | null;
  gsmParent: string | null;
  gsmBinome: string | null;
  email: string | null;
  facebook: string | null;
  tiktok: string | null;
  instagram: string | null;
  diplome: string | null;
  specialite: string | null;
  niveauScolaire: string | null;
  permisConduite: boolean;
  datePermis: string | null;
  typeContrat: string | null;
  numeroCnss: string | null;
  numeroD17: string | null;
  numeroBanquePoste: string | null;
  groupeSanguin: GroupeSanguin | null;
  poidsKg: number | null;
  hauteurCm: number | null;
  pointureChaussure: string | null;
  tailleVetements: string | null;
  maladiesChroniques: string | null;
  allergies: string | null;
  operations: string | null;
  tatouage: boolean;
  dateEmbauche: string | null;
  experience: string | null;
  societesEtPeriodes: string | null;
  salaireDepart: number | null;
  nombreJoursConge: number | null;
  gsmSociete: string | null;
  photoProfilPath: string | null;
  photoCouverturePath: string | null;
  copieCinPath: string | null;
  copieExtraitNaissancePath: string | null;
  copieDiplomePath: string | null;
  copiePermisPath: string | null;
  createdAt: string;
  updatedAt: string;
}
export interface TechnicienProfileFormData {
  nom: string;
  prenom: string;
  dateNaissance: string | null;
  cin: string | null;
  nomParent: string | null;
  adresse: string | null;
  gsm: string | null;
  gsmParent: string | null;
  gsmBinome: string | null;
  email: string | null;
  facebook: string | null;
  tiktok: string | null;
  instagram: string | null;
  diplome: string | null;
  specialite: string | null;
  niveauScolaire: string | null;
  permisConduite: boolean;
  datePermis: string | null;
  typeContrat: string | null;
  numeroCnss: string | null;
  numeroD17: string | null;
  numeroBanquePoste: string | null;
  groupeSanguin: GroupeSanguin | null;
  poidsKg: number | null;
  hauteurCm: number | null;
  pointureChaussure: string | null;
  tailleVetements: string | null;
  maladiesChroniques: string | null;
  allergies: string | null;
  operations: string | null;
  tatouage: boolean;
  dateEmbauche: string | null;
  experience: string | null;
  societesEtPeriodes: string | null;
  salaireDepart: number | null;
  nombreJoursConge: number | null;
  gsmSociete: string | null;
}